src/plugins/query_enhancements/common/utils.ts

```diff
@@ -133,7 +133,20 @@ export const handleDataFrameError = (response: any) => {
   }
 };
 
-export const fetchDataFrame = (context: FetchDataFrameContext, query: Query, df: IDataFrame) => {
+export const fetch = (context: EnhancedFetchContext, query: Query, aggConfig?: QueryAggConfig) => {
+  const { http, path, signal } = context;
+  const body = JSON.stringify({ query: { ...query, format: 'jdbc' }, aggConfig });
+  return from(
+    http.fetch({
+      method: 'POST',
+      path,
+      body,
+      signal,
+    })
+  ).pipe(tap(handleDataFrameError));
```

Comment:
```suggestion
  ).pipe(tap(handleResponseError));
```
Nit: This is more readable if that its doing is just handling the http fetch error. If its doing anything else, ignore me